### PR TITLE
Fix prometheus exporter to handle double values

### PR DESCRIPTION
**Pull request by cijothomas**
Will do cleanups/extracting common logic to reusable functions later. This PR just addresses missing functionality.

### CODE REVIEW

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptions.cs [hunk 1]
```diff
@@ -18,8 +18,8 @@
 using System.Diagnostics;
 using System.Security;
 using OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation;
-using OpenTelemetry.Trace;
 using OpenTelemetry.Metrics;
+using OpenTelemetry.Trace;
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptions.cs:22]: unrelated, but required fix as metrics branch got broken with the latest main->metrics merge and conflict resolution.

**Review by reyang — APPROVED** [review]

File: src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs [hunk 2]
```diff
@@ -72,6 +72,26 @@ public static void WriteMetricsCollection(this PrometheusExporter exporter, Stre
...
                         }
 
+                    case MetricType.DoubleSum:
+                        {
+                            builder = builder.WithType(PrometheusCounterType);
+                            foreach (var metricPoint in metric.GetMetricPoints())
+                            {
+                                var metricValueBuilder = builder.AddValue();
+                                metricValueBuilder = metricValueBuilder.WithValue(metricPoint.DoubleValue);
+                                if (metricPoint.Keys != null)
```
Comment (mic-max) [inline src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs:82]: Is this check only necessary for `Counter`s and not `Gauge`s?

Comment (cijothomas) [reply src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs:82]: Good catch! its needed for all. (I was hoping to move these to all common helpers, so don't have to repeat this check. But thats hasn't been done yet).
Feel free to submit a PR to fix the current issue. Else I'll send one. let me know if you are working on it.

Comment (cijothomas) [reply src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs:82]: #2346

**Pull request merged**